src/client.rs

```diff
@@ -195,21 +197,37 @@ async fn run_data_channel<T: Transport>(args: Arc<RunDataChannelArgs<T>>) -> Res
     let mut conn = do_data_channel_handshake(args.clone()).await?;
 
     // Forward
-    match read_data_cmd(&mut conn).await? {
+    match read_data_cmd(&mut conn.get_reliable_stream()).await? {
         DataChannelCmd::StartForwardTcp => {
-            run_data_channel_for_tcp::<T>(conn, &args.local_addr).await?;
+            match conn {
+                TransportStream::StrictlyReliable(reliable) => {
+                    run_data_channel_for_tcp::<T, T::ReliableStream>(reliable, &args.local_addr).await?;
+                }
+                TransportStream::PartiallyReliable(_, unreliable) => {
+                    run_data_channel_for_tcp::<T, T::UnreliableStream>(unreliable, &args.local_addr).await?;
```

Comment:
This doesn't seem right. After reading `quinn::Datagrams`, I see it's really just `datagram`, just like what UDP provides. And you `AsyncWrite` wrapper simply wrap `poll_write` to send a datagram. But when you do the forwarding for a TCP service, `copy_bidirectional` is called and byte stream is forwarded, in which every byte is from the application layer, not a TCP packet, and should be guaranteed to be sent to the destination, while with `T::UnreliableStream`, it can be lost.

Reply:
Ok, I must have misunderstood what the rathole client<->server data channel was carrying.

If we are ACK'ing data from the end user, then we absolutely have to ensure that the packet does indeed arrive (reliability).

But what we can still try to take advantage of is the ability to send QUIC packet **unordered**.